src/services/rocksdb/backend.rs

```diff
@@ -0,0 +1,253 @@
+// Copyright 2022 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+use std::fmt::Debug;
+use std::fmt::Formatter;
+use std::io::Error;
+use std::io::ErrorKind;
+use std::io::Result;
+use std::path::Path;
+use std::path::PathBuf;
+use std::pin::Pin;
+use std::sync::Arc;
+use std::task::ready;
+use std::task::Context;
+use std::task::Poll;
+use std::vec::IntoIter;
+
+use anyhow::anyhow;
+use async_trait::async_trait;
+use futures::future::BoxFuture;
+use futures::Future;
+use futures::Stream;
+use pin_project::pin_project;
+use rocksdb::TransactionDB;
+
+use crate::adapters::kv;
+use crate::error::new_other_backend_error;
+use crate::Accessor;
+use crate::AccessorCapability;
+use crate::Scheme;
+
+const SCAN_LIMIT: usize = 100;
+
+/// Rocksdb backend builder
+#[derive(Clone, Default, Debug)]
+pub struct Builder {
+    /// The path to the rocksdb database
+    path: Option<PathBuf>,
+    /// the working directory of the Redis service. Can be "/path/to/dir"
+    ///
+    /// default is "/"
+    root: Option<String>,
+}
+
+impl Builder {
+    pub(crate) fn from_iter(it: impl Iterator<Item = (String, String)>) -> Self {
+        let mut builder = Builder::default();
+        for (k, v) in it {
+            let v = v.as_str();
+            match k.as_ref() {
+                "path" => builder.path(v),
+                _ => continue,
+            };
+        }
+        builder
+    }
+
+    /// Set the path to the rocksdb database
+    pub fn path<P: AsRef<Path>>(&mut self, path: P) -> &mut Self {
```

Comment:
I prefer to use `&str` here to make API cleaner.

Reply:
And it's not very clean about the path.

- Should this path be a dir or a file?
- What happened if this path exist or not exist?

Reply:
> I prefer to use `&str` here to make API cleaner.

Ok. Will update.



> And it's not very clean about the path.
> 
> * Should this path be a dir or a file?
> * What happened if this path exist or not exist?

- A directory.
- A new directory will be created if it not exists.

IMO `db_dir` / `OPENDAL_ROCKSDB_DB_DIR` is clearer than `path` / `OPENDAL_ROCKSDB_PATH`, but somewhat verbose. Better naming suggestions are welcomed!

Reply:
How about OPENDAL_ROCKSDB_DATADIR？

Reply:
> How about OPENDAL_ROCKSDB_DATADIR？

Sounds better!